record-layer.go

```diff
@@ -42,72 +51,111 @@ type RecordLayer struct {
 	cachedRecord *TLSPlaintext // Last record read, cached to enable "peek"
 	cachedError  error         // Error on the last record read
 
-	ivLength int         // Length of the seq and nonce fields
-	seq      []byte      // Zero-padded sequence number
-	nonce    []byte      // Buffer for per-record nonces
-	cipher   cipher.AEAD // AEAD cipher
+	cipher   *cipherState
```

Comment:
Do you want to just make this a `map[Epoch]*cipherState`?  Then just append to it in `Rekey()` and query it on encrypt/decrypt.  Might require special handling to remove the "special" epochs for clear, handshake, etc. once they're disallowed.

Reply:
Probably in the future, but I think later.